R/mergecategoriescontainingranges.R

```diff
@@ -198,15 +214,40 @@ MergeRangeCategories <- function(input.data,
     # with user-supplied values if supplied
     if (is.na(label.data[1, "second.value"])) {
         label.data[1, "second.value"] <- label.data[1, "first.value"]
-        if (!is.null(lower.bound)) {
+        if (nzchar(lower.bound)) {
             label.data[1, "first.value"] <- as.numeric(lower.bound)
         } else {
-            label.data[1, "first.value"] <- NA
+            lower.interp <- label.data[1, "second.value"] - (label.data[2, "second.value"] - label.data[2, "first.value"])
+            if (is.na(lower.interp) | lower.interp < 0) {
+                lower.interp <- 0
+            }
+            label.data[1, "first.value"] <- lower.interp
+            if (method == "even.ranges") {
+                warning("The lower value for ", 
+                     label.data[1, "label"], 
+                     " has been set to ",
+                     lower.interp,
+                     ". To change this, enter a new value in NPUT DATA LABELS > Start of Range")    
```

Comment:
This code block very similar to the next one. Could abstract it out to a helper function that operators on the lower or upper bounds as an argument?